cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/TestCommand.java

```diff
@@ -181,13 +183,57 @@ public void execute() {
 
                 this.sourceRootPath = findRoot;
             }
+
         } else if (this.argList.get(0).endsWith(BLangConstants.BLANG_SRC_FILE_SUFFIX)) {
-            // when a single bal file is provided.
-            CommandUtil.printError(this.errStream,
-                    "test command is only supported inside a Ballerina project",
-                    null,
-                    false);
-            CommandUtil.exitError(this.exitWhenFinish);
+            // when a single bal file is provided
+            // Check if path given is an absolute path. Update the root accordingly
+            if (Paths.get(this.argList.get(0)).isAbsolute()) {
+                sourcePath = Paths.get(this.argList.get(0));
+            } else {
+                sourcePath = sourceRootPath.resolve(this.argList.get(0));
+            }
+
```

Comment:
@Ibaqu Shall we get rid of the unnecessary new lines that are introduced?

Reply:
Done